src/test/crimson/seastore/transaction_manager_test_state.h

```diff
@@ -210,9 +210,12 @@ class EphemeralTestState {
   }
 
   seastar::future<> tm_setup(
-    journal_type_t type = journal_type_t::SEGMENTED) {
+    std::string j_type = "segmented") {
     LOG_PREFIX(EphemeralTestState::tm_setup);
-    journal_type = type;
+    journal_type_t journal_type = journal_type_t::SEGMENTED;
+    if (j_type == "circularbounded") {
+      journal_type = journal_type_t::RANDOM_BLOCK;
+    }
```

Comment:
Looks `test_object_data_handler` is missing in this PR?

If all tests with `EphemeralTestState` should validate both segment and RBM, probably it's cleaner to make `EphemeralTestState` to inherit from `::testing::WithParamInterface<const char*>`, and define a helper macro `INSTANTIATE_EPHEMERAL_TESTS_P(name, SuiteType)` for reuse.